lib/stargate/receiver.ex

```diff
@@ -139,9 +139,15 @@ defmodule Stargate.Receiver do
       |> Map.merge(setup_state)
       |> (fn fields -> struct(State, fields) end).()
 
-    WebSockex.start_link(state.url, __MODULE__, state,
-      name: via(state.registry, :"sg_#{type}_#{state.tenant}_#{state.namespace}_#{state.topic}")
-    )
+    server_opts =
+      args
+      |> Stargate.Connection.auth_settings()
+      |> Keyword.put(
+        :name,
+        via(state.registry, :"sg_#{type}_#{state.tenant}_#{state.namespace}_#{state.topic}")
+      )
+
+    WebSockex.start_link(state.url, __MODULE__, state, server_opts)
```

Comment:
I'm not opposed to this but if I'm reading the quote from the Websockex docs correctly (and I don't think it's written super clearly) I would assume that means having SSL options of some kind in the `%Conn{}` means those options will take precedence when starting the link to the remote websocket server.

If that's the case, I believe you'll need to pass the desired security options in as a second argument to the `Conn.new/2` so they're included. However, if you look at websockex.ex on line 382 it should be taking the url stargate is passing and constructing a `%Conn{}`, passing the options that include `:ssl_options`, `:cacerts`, and `:extra_headers` as the second argument to `Conn.new/2` which _should_ be including them in the connection options we want. Unless I'm reading that wrong.

Reply:
You make a great point, there's a big chance that the wording means exactly that. I'm experimenting on my fork on the master branch at the moment, I will try this since I got the same error